data race on JetStream shutdown

```text
=== RUN   TestJetStreamClusterAckPendingWithMaxRedelivered
1864==================
1865WARNING: DATA RACE
1866Write at 0x00c0007179a0 by goroutine 172:
1867  github.com/nats-io/nats-server/server.(*Server).shutdownJetStream()
1868      /home/travis/gopath/src/github.com/nats-io/nats-server/server/jetstream.go:791 +0x56f
1869  github.com/nats-io/nats-server/server.(*Server).Shutdown()
1870      /home/travis/gopath/src/github.com/nats-io/nats-server/server/server.go:1758 +0x219
1871  github.com/nats-io/nats-server/server.(*cluster).shutdown()
1872      /home/travis/gopath/src/github.com/nats-io/nats-server/server/test_test.go:198 +0xd8
1873  github.com/nats-io/nats-server/server.TestJetStreamClusterAckPendingWithMaxRedelivered()
1874      /home/travis/gopath/src/github.com/nats-io/nats-server/server/jetstream_cluster_test.go:5758 +0x8d3
1875  testing.tRunner()
1876      /home/travis/.gimme/versions/go1.16.5.linux.amd64/src/testing/testing.go:1193 +0x202
1877
1878Previous read at 0x00c0007179a0 by goroutine 112:
1879  github.com/nats-io/nats-server/server.(*Server).jsConsumerInfoRequest()
1880      /home/travis/gopath/src/github.com/nats-io/nats-server/server/jetstream_api.go:3423 +0x153a
1881  github.com/nats-io/nats-server/server.(*Server).jsConsumerInfoRequest-fm()
1882      /home/travis/gopath/src/github.com/nats-io/nats-server/server/jetstream_api.go:3351 +0xc4
1883  github.com/nats-io/nats-server/server.(*jetStream).apiDispatch.func1()
1884      /home/travis/gopath/src/github.com/nats-io/nats-server/server/jetstream_api.go:657 +0xba
1885
1886Goroutine 172 (running) created at:
1887  testing.(*T).Run()
1888      /home/travis/.gimme/versions/go1.16.5.linux.amd64/src/testing/testing.go:1238 +0x5d7
1889  testing.runTests.func1()
1890      /home/travis/.gimme/versions/go1.16.5.linux.amd64/src/testing/testing.go:1511 +0xa6
1891  testing.tRunner()
1892      /home/travis/.gimme/versions/go1.16.5.linux.amd64/src/testing/testing.go:1193 +0x202
1893  testing.runTests()
1894      /home/travis/.gimme/versions/go1.16.5.linux.amd64/src/testing/testing.go:1509 +0x612
1895  testing.(*M).Run()
1896      /home/travis/.gimme/versions/go1.16.5.linux.amd64/src/testing/testing.go:1417 +0x3b3
1897  github.com/nats-io/nats-server/server.TestMain()
1898      /home/travis/gopath/src/github.com/nats-io/nats-server/server/sublist_test.go:1448 +0x384
1899  main.main()
1900      _testmain.go:2655 +0x271
1901
1902Goroutine 112 (finished) created at:
1903  github.com/nats-io/nats-server/server.(*jetStream).apiDispatch()
1904      /home/travis/gopath/src/github.com/nats-io/nats-server/server/jetstream_api.go:656 +0x624
1905  github.com/nats-io/nats-server/server.(*jetStream).apiDispatch-fm()
1906      /home/travis/gopath/src/github.com/nats-io/nats-server/server/jetstream_api.go:599 +0xc4
1907  github.com/nats-io/nats-server/server.(*client).deliverMsg()
1908      /home/travis/gopath/src/github.com/nats-io/nats-server/server/client.go:3091 +0x507
1909  github.com/nats-io/nats-server/server.(*client).processMsgResults()
1910      /home/travis/gopath/src/github.com/nats-io/nats-server/server/client.go:4076 +0x864
1911  github.com/nats-io/nats-server/server.(*client).processInboundRoutedMsg()
1912      /home/travis/gopath/src/github.com/nats-io/nats-server/server/route.go:443 +0x328
1913  github.com/nats-io/nats-server/server.(*client).processInboundMsg()
1914      /home/travis/gopath/src/github.com/nats-io/nats-server/server/client.go:3407 +0x95
1915  github.com/nats-io/nats-server/server.(*client).parse()
1916      /home/travis/gopath/src/github.com/nats-io/nats-server/server/parser.go:477 +0x3f44
1917  github.com/nats-io/nats-server/server.(*client).readLoop()
1918      /home/travis/gopath/src/github.com/nats-io/nats-server/server/client.go:1174 +0x824
1919  github.com/nats-io/nats-server/server.(*Server).createRoute.func1()
1920      /home/travis/gopath/src/github.com/nats-io/nats-server/server/route.go:1375 +0x52
1921==================
1922    testing.go:1092: race detected during execution of test
1923--- FAIL: TestJetStreamClusterAckPendingWithMaxRedelivered (1.36s)
```

--- a/server/jetstream_api.go
+++ b/server/jetstream_api.go
@@ -3377,6 +3377,7 @@ func (s *Server) jsConsumerInfoRequest(sub *subscription, c *client, subject, re
 		if ca != nil && ca.pending {
 			ca = nil
 		}
+		ourID := cc.meta.ID()
 		js.mu.RUnlock()
 
 		if isLeader && ca == nil {
@@ -3420,7 +3421,7 @@ func (s *Server) jsConsumerInfoRequest(sub *subscription, c *client, subject, re
 				// Delaying an error response gives the leader a chance to respond before us
 				s.sendDelayedAPIErrResponse(ci, acc, subject, reply, string(msg), s.jsonResponse(&resp), ca.Group)
 			} else if ca != nil {
-				if rg := ca.Group; rg != nil && rg.node != nil && rg.isMember(cc.meta.ID()) {
+				if rg := ca.Group; rg != nil && rg.node != nil && rg.isMember(ourID) {
 					// Check here if we are a member and this is just a new consumer that does not have a leader yet.
 					if rg.node.GroupLeader() == _EMPTY_ && !rg.node.HadPreviousLeader() {
 						resp.Error = ApiErrors[JSConsumerNotFoundErr]

